pkg/controller/service/servicecontroller.go

```diff
@@ -339,14 +339,23 @@ func (s *ServiceController) createLoadBalancerIfNeeded(namespacedName types.Name
 
 		// TODO: We could do a dry-run here if wanted to avoid the spurious cloud-calls & events when we restart
 
-		// The load balancer doesn't exist yet, so create it.
-		s.eventRecorder.Event(service, api.EventTypeNormal, "CreatingLoadBalancer", "Creating load balancer")
-
-		err := s.createLoadBalancer(service, namespacedName)
+		_, exists, err := s.balancer.GetLoadBalancer(s.loadBalancerName(service), s.zone.Region)
```

Comment:
Im asking: what happens if the pool contains the *wrong* host?
In the old code, we called CreateLB always, will that have fixed the wrong hosts?
If no, is there any way to detect (inside createLb) and fix this situation? If there a way we can keep this change within the openstack implementation, because I believe the gce implementation will fix the wrong host scenario, so not calling createLb will introduce a regression in other cloud providers.